test/SparqlAntlrParserTestHelpers.h

```diff
@@ -320,18 +384,17 @@ MATCHER_P4(IsSolutionModifier, groupByVariables, havingClauses, orderBy,
   };
   auto pimplComp = [](const sparqlExpression::SparqlExpressionPimpl& a,
                       const std::string& b) { return a.getDescriptor() == b; };
-  auto orderKeyComp =
-      [&exprOrderKeyComp, &equalComp, &falseComp](
-          const OrderKey& a,
-          const variant<std::pair<std::string, bool>, VariableOrderKey>& b) {
-        return std::visit(
-            ad_utility::OverloadCallOperator{exprOrderKeyComp, equalComp,
-                                             falseComp},
-            a, b);
-      };
+  auto orderKeyComp = [&exprOrderKeyComp, &equalComp, &falseComp](
+                          const OrderKey& a,
+                          const std::variant<std::pair<std::string, bool>,
+                                             VariableOrderKey>& b) {
+    return std::visit(ad_utility::OverloadCallOperator{exprOrderKeyComp,
+                                                       equalComp, falseComp},
+                      a, b);
+  };
```

Comment:
I dont really understand what you mean. This lambda is used in a `std::equal` in the return statement. The `IsSolutionModifier` is used in the `SolutionModifier` test in `SparqlAntlrParserTest`. The reason here is that the parsed objects can contain expression (OrderKey, GroupKey). They are currently tested by matching the expression to a string (descriptor).

Maybe this can be cleaned up or even transformed into a lambda when the `IsVariable...Key` a `IsExpression...Key` matchers are generalized to `Is...Key`.